activerecord/lib/active_record/connection_adapters/abstract/database_statements.rb

```diff
@@ -359,6 +359,22 @@ def transaction(requires_new: nil, isolation: nil, joinable: true, &block)
           end
           yield current_transaction.user_transaction
         else
+          # If we're running inside the single, non-joinable transaction that
+          # ActiveRecord::TestFixtures starts around each example (depth == 1),
+          # an `isolation:` hint must be validated then ignored so that the
+          # adapter isn't asked to change the isolation level mid-transaction.
+          if isolation && open_transactions == 1 && !current_transaction.joinable?
+            iso = isolation.to_sym
+
+            unless transaction_isolation_levels.include?(iso)
+              raise ActiveRecord::TransactionIsolationError,
+                    "invalid transaction isolation level: #{iso.inspect}"
+            end
+
+            current_transaction.isolation = iso
```

Comment:
Is this better placed inside `within_new_transaction`? That already has light `isolation` handling, and it feels slightly less bad layer-breaking to keep it in TransactionManager, maybe?

Should we also restore the previous isolation at the end of the block? Does leaving a false residual isolation level potentially create surprises when interacting with other later transaction blocks?